sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveShim.scala

```diff
@@ -111,6 +112,31 @@ private[hive] object HiveShim {
     }
   }
 
+  private def isSubClassOf(t: Type, parent: Class[_]): Boolean = t match {
+    case cls: Class[_] => parent.isAssignableFrom(cls)
+    case _ => false
+  }
+
+  private def hasInheritanceOf[UDFType: ClassTag](funcName: String, clazz: Class[_]): Boolean = {
+    val clsTag = classTag[UDFType].runtimeClass
+    if (isSubClassOf(clazz, clsTag)) {
+      val funcClass = clazz.getMethod(funcName, classOf[MapredContext])
+      funcClass.getDeclaringClass != clsTag
+    } else {
+      false
+    }
+  }
+
+  def validateHiveUserDefinedFunction(udfClass: Class[_]): Unit = {
+    if (hasInheritanceOf[GenericUDF]("configure", udfClass) ||
```

Comment:
IIUC spark always refers to `hive-exec-1.2.1.spark2.jar `, so it seems we have no chance to get the exception. But, I think it is not a bad idea to catch `NoSuchMethodException` there for understandability.